src/Tags.elm

```diff
@@ -0,0 +1,175 @@
+                  , Html.Attributes.class "text-sm font-semibold text-indigo-600"
+                  , Html.Attributes.href "#"
+                  ]
+                  [ Html.text "+ Add new tag" ]
+            ]
+        , Html.button
+              [ Html.Attributes.class "rounded-md bg-indigo-600 px-3 py-2 text-sm font-semibold text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600 disabled:bg-slate-400 disabled:text-slate-50 disabled:border-slate-200 disabled:shadow-none"
+              ]
+              [ Html.text "Save" ]
+        ]
+
+
+tagView : String -> Form () Tags -> Int -> Html Form.Msg
+tagView path form idx =
+    let
+        prefix =
+            path ++ "." ++ (String.fromInt idx)
+            
+        nameField =
+            Form.getFieldAsString (prefix ++ ".name") form
+
+        valueField =
+            Form.getFieldAsString (prefix ++ ".value") form
+    in
+        Html.div [ Html.Attributes.class "sm:col-span-full" ]
+            [ Html.div [ Html.Attributes.class "relative flex gap-x-3 items-top"]
+                  [ textField (prefix ++ ".name") "Name" form
+                  , textField (prefix ++ ".value") "Value" form
+                  , Html.div [ Html.Attributes.class "py-3" ]
+                      [ Html.a
+                            [ Html.Events.onClick (Form.RemoveItem path idx)
+                            , Html.Attributes.class "text-sm font-semibold text-indigo-600"
+                            , Html.Attributes.href "#"
+                            ]
+                            [ Html.text "Remove" ]
+                      ]
+                  ]
+            ]
+
+
+textField : String -> String -> Form () Tags -> Html Form.Msg
+textField path label form =
+    let
+        field =
+            Form.getFieldAsString path form
+    in
+        Html.div [ Html.Attributes.class "sm:col-span-full"]
+            [ Html.div [ Html.Attributes.class "mt-2"]
+                [ Input.textInput field
+                      [ Html.Attributes.id field.path
+                      , Html.Attributes.placeholder label
+                      , Html.Attributes.class "block w-full rounded-md border-0 px-2 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6 disabled:bg-slate-50 disabled:text-slate-500 disabled:border-slate-200 disabled:shadow-none"
+                      , case field.liveError of
+                            Just _ ->
+                                Html.Attributes.class "bg-red-50 border border-red-500 text-red-900 placeholder-red-700 text-sm rounded-lg focus:ring-red-500 focus:border-red-500 block w-full p-2.5 dark:bg-red-100 dark:border-red-400"
+                          
+                            Nothing ->
+                                Html.Attributes.class ""
+                      ]
```

Comment:
NIT: I would probably extract this into `let..in` block because I don't like to mix branching logic and view code together much. Also I might use some helpers from [`Html.Attribites.Extra`](https://package.elm-lang.org/packages/elm-community/html-extra/3.4.0/#attributeMaybe)

Reply:
Yeah, I think it really should live in its own helper function. Not just in the `let..in` block, but as its own function. Because the same, or very similar, code is used in several of the view functions.